geth/api/api.go

```diff
@@ -121,21 +124,46 @@ func (api *StatusAPI) CallRPC(inputJSON string) string {
 // BIP44-compatible keys are generated: CKD#1 is stored as account key, CKD#2 stored as sub-account root
 // Public key of CKD#1 is returned, with CKD#2 securely encoded into account key file (to be used for
 // sub-account derivations)
-func (api *StatusAPI) CreateAccount(password string) (address, pubKey, mnemonic string, err error) {
-	return api.b.AccountManager().CreateAccount(password)
+func (api *StatusAPI) CreateAccount(password string) common.AccountInfo {
+	address, pubKey, mnemonic, err := api.b.AccountManager().CreateAccount(password)
+	errString := processError(err)
+	return common.AccountInfo{
```

Comment:
Handling errors in a Go-way would look like this:

A new type that implements `error` and `Marshaler` interfaces are needed.
```go
type AccountError struct {
	err error
}

func NewAccountError(err error) *AccountError {
 	return &AccountError{err}
}

func (e *AccountError) Error() string {
	return fmt.Sprintf("account error: %s", e.err.Error())
}

func (e *AccountError) MarshalJSON() ([]byte, error)  {
 	return json.Marshal(e.err.Error())
}
```

Usage:
```go
return common.AccountInfo{
  // other fields here
  Error: common.NewAccountError(err),
}

data, _ := json.Marshal(&err)
// string(data) equals: { "error": "some error" }
```

Reply:
Would this be instead of returning err as a separate return value as per themue comment above? Or are you suggesting including AccountError in AccountInfo instead of AccountInfo.Error as well as returning err as a separate value?

Reply:
Actually, @themue's idea is very good. I guess we can return an error as a separate value and use it like this:
```go
accountInfo, err := statusAPI.CreateAccount(string)
if err != nil {
  return makeJSONResponse(err)
}

// marshal accountInfo and return it
```

The only difference is that `AccountInfo` won't contain `Error` field. I am not sure if status-react is prepared for this change. I will figure out that and let you know.

Reply:
@jonprice Let's proceed with @themue's idea to return an error as a second argument.

Reply:
Made the change to return error as a second return param, but left the lib/library.go to return the accountInfo with Error value. 

I did some investigation in status-react to see if I could work out how the AccountInfo.Error was handled. I can't see any code that handles the error message, so I'm not sure if Clojure makes the error handling look transparent, or if there isn't any handling of the error case. Either way, I could not determine the impact of changing how the error case was returned.